pkg/core/bundle_list.go

```diff
@@ -12,52 +13,128 @@ import (
 )
 
 const (
-	maxBundlesToList = 1000000
+	maxBundlesToList  = 1000000
+	typicalBundlesNum = 1000
 )
 
+type bundleChan struct {
+	bundle model.BundleDescriptor
+	err    error
+}
+
+type bundlesChan struct {
+	bundles []model.BundleDescriptor
+	err     error
+}
+
+// ListBundles returns the list of bundle descriptors from a repo.
+//
 // TODO: return a paginated list of id<->bd (separate function in repo_list.go)
-func ListBundles(repo string, store storage.Store) ([]model.BundleDescriptor, error) {
+func ListBundles(repo string, store storage.Store, opts ...BundleOption) ([]model.BundleDescriptor, error) {
+	var concurrencyFactor int
+	b := defaultBundle()
+	for _, bApply := range opts {
+		bApply(&b)
+	}
+	concurrencyFactor = b.concurrentBundleList
+
 	// Get a list
 	e := RepoExists(repo, store)
 	if e != nil {
 		return nil, e
 	}
+
+	// this call is not made async yet
 	ks, _, err := store.KeysPrefix(context.Background(), "", model.GetArchivePathPrefixToBundles(repo), "", maxBundlesToList)
 	if err != nil {
 		return nil, err
 	}
-	bds := make([]model.BundleDescriptor, 0)
-	for _, k := range ks {
+
+	var wg sync.WaitGroup
+
+	result := make(chan bundlesChan)
+	bundle := make(chan bundleChan, concurrencyFactor)
```

Comment:
in light of the comment on the channel value naming, could we call this `bundleC`, `bundleChan` or something?